Mathlib/SetTheory/Cardinal/Finite.lean

```diff
@@ -149,4 +149,123 @@ theorem card_eq_top_of_infinite [Infinite α] : card α = ⊤ :=
   mk_toPartENat_of_infinite
 #align part_enat.card_eq_top_of_infinite PartENat.card_eq_top_of_infinite
 
+theorem card_congr {α : Type _} {β : Type _} (f : α ≃ β) : PartENat.card α = PartENat.card β :=
+  Cardinal.toPartENat_congr f
+#align part_enat.card_congr PartENat.card_congr
+
+theorem card_uLift (α : Type _) : card (ULift α) = card α :=
+  card_congr Equiv.ulift
+#align part_enat.card_ulift PartENat.card_uLift
+
+@[simp]
+theorem card_pLift (α : Type _) : card (PLift α) = card α :=
+  card_congr Equiv.plift
+#align part_enat.card_plift PartENat.card_pLift
+
+theorem card_image_of_injOn {α : Type u} {β : Type v} {f : α → β} {s : Set α} (h : Set.InjOn f s) :
+    card (f '' s) = card s :=
+  card_congr (Equiv.Set.imageOfInjOn f s h).symm
+#align part_enat.card_image_of_inj_on PartENat.card_image_of_injOn
+
+theorem card_image_of_injective {α : Type u} {β : Type v} (f : α → β) (s : Set α)
+    (h : Function.Injective f) : card (f '' s) = card s :=
+  card_congr (Equiv.Set.imageOfInjOn f s (Set.injOn_of_injective h s)).symm
+#align part_enat.card_image_of_injective PartENat.card_image_of_injective
+
+-- Add other similar lemmas?
+theorem succ_le_iff {n : ℕ} {e : PartENat} : ↑n.succ ≤ e ↔ ↑n < e := by
```

Comment:
Moved, renamed to `coe_succ_le_iff_lt` and added its companion `lt_coe_succ_iff_le`

Reply:
The proof uses the preexisting `add_one_le_iff_lt` and `lt_add_one_iff_le`. 
Question : there are also direct lemmas `add_one_le_of_lt (proved before). Should I write their `coe_succ` versions?  (Just by symmetry, because `rw` suffices…)